libs/community/langchain_community/vectorstores/cloudSQL.py

```diff
@@ -0,0 +1,330 @@
+        embedding_service: Embeddings, 
+        table_name: str, 
+        content_column: str='content', 
+        embedding_column: str='embedding', 
+        metadata_columns: Optional[str, List[str]]='metadata', 
+        ignore_metadata_columns: Optional[str, List[str]]=None, 
+        overwrite_existing: bool=False, 
+        index_query_options=None, 
+        distance_strategy: str="L2"
+    ) -> None:
+        """Constructor for CloudSQLVectorStore.
+
+        Args:
+            engine (sqlalchemy.engine.Engine): Sqlalchmey engine with pool connection to the postgres database. Required.
+            embedding_service (Embeddings): Text embedding model to use.
+            table_name (str): Name of the existing table or the table to be created.
+            content_column (str): Column that represent a Document’s page_content. Defaults to content
+            embedding_column (str): Column for embedding vectors. 
+                              The embedding is generated from the document value. Defaults to embedding
+            metadata_columns (List[str]): Column(s) that represent a document's metdata. Defaults to metdata
+            ignore_metadata_columns (List[str]): Column(s) to ignore in pre-existing tables for a document’s metadata. 
+                                     Can not be used with metadata_columns. Defaults to None
+            overwrite_existing (bool): Boolean for truncating table before inserting data. Defaults to False
+            index_query_options : QueryOptions class with vector search parameters. Defaults to None
+            distance_strategy (str):
+                Determines the strategy employed for calculating
+                the distance between vectors in the embedding space.
+                Defaults to EUCLIDEAN_DISTANCE(L2).
+                Available options are:
+                - COSINE: Measures the similarity between two vectors of an inner
+                    product space.
+                - EUCLIDEAN_DISTANCE: Computes the Euclidean distance between
+                    two vectors. This metric considers the geometric distance in
+                    the vector space, and might be more suitable for embeddings
+                    that rely on spatial relationships. This is the default behavior.
+            """
+
+        self.engine = engine
+        self.embedding_service = embedding_service
+        self.table_name = table_name
+        self.content_column = content_column
+        self.embedding_column = embedding_column
+        self.metadata_columns = metadata_columns
+        self.ignore_metadata_columns = ignore_metadata_columns
+        self.overwrite_existing = overwrite_existing
+        self.index_query_options = index_query_options
+        self.distance_strategy = distance_strategy
+        self.__post_init__()
+        
+    def __post_init__(self) -> None:
+        """Initialize table and validate existing tables"""
+        self.create_vector_extension()
```

Comment:
I think we probably want to do this as part of the `init_vectorstore_table` method.